ra/ra.go

```diff
@@ -1700,7 +1700,7 @@ func revokeEvent(state, serial, cn string, names []string, revocationCode revoca
 
 // revokeCertificate generates a revoked OCSP response for the given certificate, stores
 // the revocation information, and purges OCSP request URLs from Akamai.
-func (ra *RegistrationAuthorityImpl) revokeCertificate(ctx context.Context, cert *x509.Certificate, reason revocation.Reason, revokedBy int64, source string, comment string) error {
+func (ra *RegistrationAuthorityImpl) revokeCertificate(ctx context.Context, cert *x509.Certificate, reason revocation.Reason, revokedBy int64, source string, comment string, skipBlockKey bool) error {
```

Comment:
Is it worth adding a check that throws an error if you try to pass `skipBlockKey=true` when the `reason` is something other than `keyCompromise`? That would help catch any code which incorrectly tries to skip blocking by default.

Reply:
Added the check on `1851`